skore/src/skore/sklearn/train_test_split/warning/time_based_column_warning.py

```diff
@@ -0,0 +1,63 @@
+"""'Time-based column' warning.
+
+This warning is shown when ``X`` contains a time-based column.
+"""
+
+from __future__ import annotations
+
+import re
+from typing import TYPE_CHECKING, Any, Union
+
+from skore.sklearn.train_test_split.warning.train_test_split_warning import (
+    TrainTestSplitWarning,
+)
+
+if TYPE_CHECKING:
+    ArrayLike = Any
+
+
+class TimeBasedColumnWarning(TrainTestSplitWarning):
+    """Check whether the design matrix ``X`` contains a time-based column."""
+
+    @staticmethod
+    def _MSG(df_name: str, offending_column_names: list[str]) -> str:
+        s = "" if len(offending_column_names) == 1 else "s"
+
+        df_name_info = "" if df_name is None else f', dataframe "{df_name}"'
+
+        column_info = f"(column{s} {', '.join(offending_column_names)}{df_name_info})"
+
+        return (
+            f"We detected some time-based columns {column_info} in your data. "
+            "We recommend using scikit-learn's TimeSeriesSplit instead of "
+            "train_test_split. Otherwise you might train on future data to "
+            "predict the past, or get inflated model performance evaluation "
+            "because natural drift will not be taken into account."
+        )
+
+    @staticmethod
+    def check(X: ArrayLike, **kwargs) -> Union[str, None]:
+        """Check whether the design matrix ``X`` contains a time-based column.
+
+        Parameters
+        ----------
+        X : array-like or None
+            A data matrix that can contain datetime data, e.g. a DataFrame.
+
+        Returns
+        -------
+        warning
+            None if the check passed, otherwise the warning message.
+        """
+        if not hasattr(X, "columns"):
+            return None
+
+        dtypes = [(col, X[col].dtype) for col in X.columns]
+        datetime_columns = [
+            f'"{col}"'
+            for col, dtype in dtypes
+            if re.search("date", str(type(dtype)), flags=re.IGNORECASE)
+        ]
```

Comment:
Why not just checking for type? Something like:

```python
datetime_columns = [
    str(column)
    for column in X.columns
    if X[column].dtype in (datetime.datetime, datetime.date, polars.Date, polars.Datetime)
]
```

Reply:
That would require adding `polars` as a dependency of skore

Reply:
Please add a comment to describe your choice.